tpl/data/data_test.go

```diff
@@ -119,6 +119,20 @@ func TestGetCSV(t *testing.T) {
 		c.Assert(got, qt.Not(qt.IsNil), msg)
 		c.Assert(got, qt.DeepEquals, test.expect, msg)
 
+		// Test user-defined headers as well
+		gotHeader, _ := ns.GetCSV(test.sep, test.url, map[string]interface{}{"Accept-Charset": "utf-8", "Max-Forwards": "10"})
```

Comment:
To improve these header tests I suggest that we just echo these headers in the response and then add an assertion for one of them. 

Maybe we could also add the URL in the response as a custom header and assert on that? That way we can make sure that that is well formed.